internal/guest/storage/ext4/format.go

```diff
@@ -0,0 +1,27 @@
+//go:build linux
+// +build linux
+
+package ext4
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+)
+
+// mkfsExt4Command runs mkfs.ext4 with the provided arguments
+func mkfsExt4Command(args []string) error {
+	cmd := exec.Command("mkfs.ext4", args...)
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("failed to execute mkfs.ext4: %s: %w", string(output), err)
+	}
+	return nil
+}
+func FormatDiskExt4(ctx context.Context, source string) error {
```

Comment:
Is the idea is to have a separate package per fs if we need others in the future?

Reply:
The idea is any ext4 based operations will be in this package. If there are other fs operations then yes, they'd go in another package. We can reevaluate if they need to be in separate packages later if it doesn't seem necessary.